api/provider/aws/deploy_create.go

```diff
@@ -47,6 +47,46 @@ func (d *DeployProvider) createTaskDefinition(taskDefinition *ecs.TaskDefinition
 		input.SetNetworkMode(nm)
 	}
 
+	// We'll be explicit here and set any and all compatibilities that a user specifies in the task definition.
+	// At the moment, that should only be "FARGATE" and "EC2".
+	requiresCompatibilities := []*string{}
+	for _, compatibility := range taskDefinition.RequiresCompatibilities {
+		requiresCompatibilities = append(requiresCompatibilities, compatibility)
+
+		// There are some additional requirements for a task definition to be considered Fargate-compatible:
+		// https://github.com/aws/aws-sdk-go/blob/master/service/ecs/api.go#L8172
+		// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size
+		//
+		// For the most part, there are four things that should be specified in a Fargate task definition
+		// that aren't required for an EC2 task definition:
+		//   "requiresCompatibilities": [ "FARGATE" ],
+		//   "networkMode": "awsvpc",
+		//   "cpu": "<an appropriate value>",
+		//   "memory": "<an appropriate value>"
+		if aws.StringValue(compatibility) == ecs.LaunchTypeFargate {
+			cpu := aws.StringValue(taskDefinition.Cpu)
+			memory := aws.StringValue(taskDefinition.Memory)
+
+			// Question: Do we want to be proactive here and error for certain configurations of a task
+			// definition that we know to be bad? They would come out of AWS when the user tried to use the
+			// deploy in a service or a task, but it might be nice UX to save them a step.
```

Comment:
👍 very much agree w this logic

Reply:
(This is relevant to [this comment](https://github.com/quintilesims/layer0/pull/533#discussion_r170122962), just above.)